manim/scene/streaming_scene.py

```diff
@@ -0,0 +1,121 @@
+from ..mobject.frame import FullScreenRectangle as Frame
+from ..renderer.stream_renderer import StreamCairoRenderer
+from ..utils.simple_functions import get_parameters
+from .scene import Scene
+
+
+class Stream:
+    """This class is really intended for inheritance of the style:
+
+    >>> class Streamer(Stream, Scene): # doctest: +SKIP
+    ...     pass
+    ...
+    >>>
+
+    This order is paramount. This :class:`Stream` class does the switch to
+    the specialized renderer, which uses the :class:`StreamFileWriter` to
+    handle specialized streaming services. That explains the calls to super,
+    which dig through the MRO of a class instead of using just a single
+    implementation contained in Scene. Okay, the references in super probably
+    point to things in the :class:`Scene` class only, but it's already happened so...
+
+    PS: You can probably already tell using this class on its own will bring you
+    errors.
+
+    """
+
+    def __init__(self, **kwargs):
+        # TODO: Someday, when this is accepted into the community, work on camera
+        # qualities that can be set from this initialization
+        camera_class = self.mint_camera_class()
+        renderer = StreamCairoRenderer(camera_class=camera_class)
+        super().__init__(renderer=renderer, **kwargs)
+        # To identify the frame in a black background
+        self.add(Frame())
+        # TODO: What happens when setup actually has play arguments?
+        self.setup()
+
+    @classmethod
+    def mint_camera_class(cls):
+        """Only __init__ arguments have the camera class now. In order for the
+        specialized renderer to be used, the scene's camera class must be found.
+
+        Returns:
+            A camera class from the scene's inheritance hierachy
+
+        Raises:
+            AttributeError: If this lookup fails
+        """
+        for obj in cls.mro():
+            try:
+                parameter = get_parameters(obj.__init__)["camera_class"]
+            except KeyError:
+                continue
+            else:
+                return parameter.default
+        raise AttributeError("Object does not contain scene protocol")
+
+    def show_frame(self):
+        """
+        Opens the current frame in the Default Image Viewer
+        of your system.
+        """
+        self.renderer.update_frame(self, ignore_skipping=True)
+        self.renderer.camera.get_image().show()
+
+
+def get_streamer(*scene):
+    """Creates an instance of a class that has streaming services.
+
+    Optional arguments:
+        scene: The scene whose methods can be used in the resulting
+        instance, such as zooming in and arbitrary method constructions.
+        Defaults to just Scene
+
+    Returns:
+        StreamingScene: It's a Scene that Streams. Name deconstruction.
+
+    Note:
+        Using starred positional args makes the tuple of the passed scene
+        for me. At the moment I figure multiple inheritance is extremely
```

Comment:
first person ref. can be added as further comments to PR. modify/remove before merge